tests/testthat/test-mmrm-methods.R

```diff
@@ -180,3 +180,281 @@ test_that("confint give same result as SAS on ar1", {
+                                        get_mmrm(),
+                                        any_reml = FALSE),
+                 regexp = "identical")
+  expect_true(h_assert_nested_models(get_mmrm_kr(),
+                                     get_mmrm(),
+                                     any_reml = FALSE))
+})
+
+test_that("h_assert_lrt_suitability() ensures suitability for LRT testing", {
+  # non-increasing degrees of freedom
+  expect_error(h_assert_lrt_suitability(list(get_mmrm_cs(), get_mmrm_smaller_data()),
+                                        refit = FALSE,
+                                        dfs = c(3, 3),
+                                        is_reml = c(TRUE, TRUE)),
+               regexp = "degrees of freedom")
+  # refit = FALSE and they don't use the same data
+  expect_error(h_assert_lrt_suitability(list(get_mmrm_cs(), get_mmrm_smaller_data()),
+                                        refit = FALSE,
+                                        dfs = c(attr(logLik(get_mmrm_cs()), "df"),
+                                                attr(logLik(get_mmrm_smaller_data()), "df")),
+                                        is_reml = c(component(get_mmrm_cs(), "reml"),
+                                                    component(get_mmrm_smaller_data(), "reml"))),
+               regexp = "same data")
+  expect_true(h_assert_lrt_suitability(list(get_mmrm_cs(), get_mmrm()),
+                                       refit = FALSE,
+                                       dfs = c(attr(logLik(get_mmrm_cs()), "df"), attr(logLik(get_mmrm()), "df")),
+                                       is_reml = c(component(get_mmrm_cs(), "reml"), component(get_mmrm(), "reml"))))
+})
+
+test_that("h_generate_new_name() generates a string without a binding in env", {
+  expect_equal(h_generate_new_name("c", asNamespace("stats")), "c.1")
+  this_environment <- environment()
+  test_env_parent <- new.env()
+  test_env_child <- new.env(parent = test_env_parent)
+  this_environment[["foo_mmrm_test_name.2"]] <- NA
+  test_env_parent[["foo_mmrm_test_name"]] <- NA
+  test_env_child[["foo_mmrm_test_name.1"]] <- NA
+  expect_equal(
+    h_generate_new_name("foo_mmrm_test_name", test_env_child),
+    "foo_mmrm_test_name.3"
+  )
+})
+
+test_that("h_refit_mmrm() successfully refits an mmrm fit", {
+  fit <- get_mmrm()
+  refit <- h_refit_mmrm(get_mmrm_smaller_data(), fev_data)
+
+  # Get rid of tmb_object and call's data argument, which will be different
+  fit$tmb_object <- NULL
+  fit$call$data <- NULL
+  refit$tmb_object <- NULL
+  refit$call$data <- NULL
+
+  expect_equal(refit, fit)
+})
+
+test_that("h_anova_single_mmrm_model() yields the significance of each term", {
+  expect_equal(
+    h_anova_single_mmrm_model(get_mmrm_group()),
+    data.frame(
+      num_df = c(1, 1),
+      denom_df = c(93.50361, 146.55428),
+      f_stat = c(10170.36271, 30.09819),
+      p_val = c(3.319442e-97, 1.757614e-7),
```

Comment:
for these kind of tests here please consistently use large enough tolerance e.g. 1e-3 because otherwise this will be a problem in CRAN submission where other machines will produce slightly different results